COMPONENTS_DOCUMENTATION_GUIDELINES.md

```diff
@@ -0,0 +1,110 @@
+        <Button size={Button.sizes.SMALL}>Small</Button>
+    </Story>
+</Canvas>
+```
+
+4. When laying out components, utilize Vibe's Flex component instead of implementing your own flex layout using CSS. Refer to the code example below for clarification. **Please use Flex only when displaying multiple examples** in a story and only if the default story layout and gaps do not match the documentation design without using Flex as a wrapper.
+```js
+<Flex direction={Flex.directions.COLUMN} gap={Flex.gaps.MEDIUM}>
+  <Steps type={Steps.types.NUMBERS} steps={steps} activeStepIndex={2} />
+  <Steps type={Steps.types.GALLERY} steps={steps} activeStepIndex={2} />
+</Flex>
+```
+
+4. If you need to add titles to the component states displayed in a story, please use the VisualDescription component, as shown in the code example below:
+```js
+<Flex direction={Flex.directions.ROW}>
+    <StoryDescription description="Xs">
+      <Loader size={Loader.sizes.XS} />
+    </StoryDescription>
+    <StoryDescription description="Small">
+      <Loader size={Loader.sizes.SMALL} />
+    </StoryDescription>
+</Flex>
+```
+6. For better story readability, it's best to use inline styles instead of class names when applying styles that are specific to the story itself. This avoids confusion for readers who might question the purpose of the class names in achieving the displayed result. Inline styles make the styles visible within the story, without the need for external CSS files or class definitions.
+```js
+<Canvas>
+  <Story name="Stretched">
+    <div style={{ width: "100%" }}>
+      <TabList tabType="stretched">
+        <Tab>First</Tab>
+        <Tab>Second</Tab>
+      </TabList>
+    </div>
+  </Story>
+</Canvas>
+```
+7. To keep our storybook code examples clear and simple, avoid using internal components (components which not exported to external users) in stories. Code duplication is acceptable if it means excluding internal components. This prevents confusion among developers using our components API. Frequent reliance on internal components for documentation suggests that the API lacks easy solutions for common use cases.  For example, instead of writing this:
+```js
+<Canvas>
+  <Story name="Default">
+    <Tipseen
+      content={
+          <TipsenStoryookContentExample />
+      }
+    >
+      <TipsenStoryookContainerExample />
+    </Tipseen>
+  </Story>
+</Canvas>
+```
+
+Do this:
+```js
+<Canvas>
+  <Story name="Default">
+    <Tipseen
+      content={
+        <TipseenContent title="This is a title" isDismissHidden>
+          Popover message will appear here loremipsum dolor samet…
+        </TipseenContent>
+      }
+    >
+      <div className="monday-storybook-tipseen_container" />
+    </Tipseen>
+  </Story>
+</Canvas>
+```
+
+8. The storybook generated code should be easily "copy-paste"-able for seamless integration. Avoid adding internal additions that require additional modifications for successful copy-pasting.
```

Comment:
Let's add another one
9. As a general rule of thumb - Never apply custom styles in a story example to make a component look good. If you need to do it, it probably means there's an issue with the component itself.